lock.c

```diff
@@ -101,3 +109,55 @@ void iio_cond_signal(struct iio_cond *cond)
 	pthread_cond_signal(&cond->cond);
 #endif
 }
+
+#ifndef NO_THREADS
+static void * iio_thrd_wrapper(void *d)
+{
+	struct iio_thrd *thrd = d;
+
+	return (void *)(intptr_t) thrd->func(thrd->d);
+}
+#endif
+
+struct iio_thrd * iio_thrd_create(int (*thrd)(void *),
+				  void *d, const char *name)
+{
+	struct iio_thrd *iio_thrd;
+	int ret;
+
+	iio_thrd = malloc(sizeof(*iio_thrd));
+	if (!iio_thrd)
+		return NULL;
+
+	iio_thrd->d = d;
+	iio_thrd->func = thrd;
```

Comment:
I know it does not make much sense for anyone calling this with `thrd` set to `NULL`  and this would be something (most likely) caught right away but I would still make sure `thrd` in not NULL and return `ERR_PTR(-EINVAL) ` (or something like that) in case it is... Having segfaults is never neat

Reply:
Alright.